src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestContextInformation.cs

```diff
@@ -14,292 +14,298 @@
             {
-                if (!VsTests.ContainsKey(testCase.Id))
-                {
-                    VsTests[testCase.Id] = new VsTestDescription(testCase);
-                }
-
-                VsTests[testCase.Id].AddSubCase();
-                _logger.LogTrace(
-                    $"Test Case : name= {testCase.DisplayName} (id= {testCase.Id}, FQN= {testCase.FullyQualifiedName}).");
+                VsTests[testCase.Id] = new VsTestDescription(testCase);
             }
 
-            DetectTestFrameworks(VsTests.Values);
-            Tests.RegisterTests(VsTests.Values.Select(t => t.Description));
+            VsTests[testCase.Id].AddSubCase();
+            _logger.LogTrace(
+                $"Test Case : name= {testCase.DisplayName} (id= {testCase.Id}, FQN= {testCase.FullyQualifiedName}).");
         }
 
-        private void DetectTestFrameworks(ICollection<VsTestDescription> tests)
-        {
-            if (tests == null)
-            {
-                _testFramework = 0;
-                return;
-            }
+        DetectTestFrameworks(VsTests.Values);
+        Tests.RegisterTests(VsTests.Values.Select(t => t.Description));
+    }
 
-            if (tests.Any(testCase => testCase.Framework == TestFrameworks.NUnit))
-            {
-                _testFramework |= TestFrameworks.NUnit;
-            }
+    private void DetectTestFrameworks(ICollection<VsTestDescription> tests)
+    {
+        if (tests == null)
+        {
+            _testFramework = 0;
+            return;
+        }
 
-            if (tests.Any(testCase => testCase.Framework == TestFrameworks.xUnit))
-            {
-                _testFramework |= TestFrameworks.xUnit;
-            }
+        if (tests.Any(testCase => testCase.Framework == TestFrameworks.NUnit))
+        {
+            _testFramework |= TestFrameworks.NUnit;
+        }
 
-            if (tests.Any(testCase => testCase.Framework == TestFrameworks.MsTest))
-            {
-                _testFramework &= ~TestFrameworks.MsTest;
-            }
+        if (tests.Any(testCase => testCase.Framework == TestFrameworks.xUnit))
+        {
+            _testFramework |= TestFrameworks.xUnit;
         }
 
-        private string GenerateRunSettingsForDiscovery()
+        if (tests.Any(testCase => testCase.Framework == TestFrameworks.MsTest))
         {
-            var testCaseFilter = string.IsNullOrWhiteSpace(Options.TestCaseFilter)
-                ? string.Empty
-                : $"<TestCaseFilter>{Options.TestCaseFilter}</TestCaseFilter>" + Environment.NewLine;
+            _testFramework &= ~TestFrameworks.MsTest;
+        }
+    }
+
+    private string GenerateCoreSettings(int maxCpu, string frameworkVersion, string platform)
+    {
+        var frameworkConfig = string.IsNullOrWhiteSpace(frameworkVersion)
+            ? string.Empty
+            : $"<TargetFrameworkVersion>{frameworkVersion}</TargetFrameworkVersion>" + Environment.NewLine;
```

Comment:
We used to set the TargetFrameworkVersion, at some point we removed this feature because VsTest can determine the correct TargetFramework by itself and we kept running into issues where we were choosing the wrong TargetFramework.

Is this really neccesary?

The runsettings documentation specifies that targetframeworkversion and targetplatform are auto detected from the built binaries. Since we now build with the correct targetframeworkversion and targetplatform through Configuration, I would assume that we should let the auto detect do it's work. See: https://learn.microsoft.com/en-us/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file?view=vs-2022

Reply:
If it saves time I think it's worthwhile, we can always revert when we get reports of failures.